ledger/src/shred.rs

```diff
@@ -592,17 +594,11 @@ impl From<ShredCode> for Shred {
     }
 }
 
-impl From<ShredData> for Shred {
-    fn from(shred: ShredData) -> Self {
-        Self::ShredData(shred)
-    }
-}
```

Comment:
Similar to my other comment about `Self::ShredData(shred)`, I think we should keep the `ShredData` and `ShredCode` paths similar. So, my vote would be to leave this `From` impl around for now to make it match.

Generally, I agree that the `From` impl's might be overkill but we could tackle that in a follow-on.